WorkspaceServer.Tests/Kernel/CompositeKernelTests.cs

```diff
@@ -163,7 +163,7 @@ public async Task Kernel_can_be_interacted_using_kernel_client()
             var reader = new StreamReader(output, Encoding.UTF8);
 
             var text = reader.ReadToEnd();
-            var events = text.Split("\r\n")
+            var events = text.Split(Environment.NewLine)
```

Comment:
We might be running on a different system from where the input was sent, or a notebook authored on one OS might be opened on another, so we should split on both `\n` and `\r\n`. I think we have code elsewhere doing that and it might be a worthwhile utility method to pull out.

Reply:
Unrelated, what if the `SubmitCode.Code` contains newlines?

Reply:
A single command/response must not contain newlines because we read them by calling console.ReadLine() (effectively). The code we're looking at here is the test code, which just pushes everything into a single stream and then reads it all back at once (hence the splitting)

Reply:
What's the process when someone submits a cell with multiple lines of code inside it? If we read the input to the end, instead of line by line, could it simplify this case?

Reply:
I think it's not valid json to have an unescaped line break in a string. It's fair for us to expect clients to escape their newlines.

Reply:
Will add a test for this though